src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.h

```diff
@@ -8,38 +8,51 @@ struct EnAob01;
 
 typedef void (*EnAob01ActionFunc)(struct EnAob01*, PlayState*);
 
-#define ENAOB01_GET_7E00_1(thisx) ((thisx)->params & 0x7E00)
-#define ENAOB01_GET_7E00_2(thisx) (((thisx)->params & 0x7E00) >> 9)
+#define ENAOB01_GET_STARTING_DOG_PATH_INDEX(thisx) (((thisx)->params & 0x7E00) >> 9)
+
+//! @bug This should be shifted up by 1 to match how EnRacedog expects the path index to be supplied.
+//! It doesn't cause any problems in the final game because every time EnAob01 is spawned, all the
+//! bits here are zero.
+#define ENAOB01_GET_RACEDOG_PATH_INDEX(thisx) ((thisx)->params & 0x7E00)
+
+#define ENAOB01_FLAG_PLAYER_CONFIRMED_CHOICE (1 << 1)
+#define ENAOB01_FLAG_PLAYER_TOLD_TO_PICK_A_DOG (1 << 2)
+#define ENAOB01_FLAG_TALKING_TO_PLAYER_HOLDING_DOG (1 << 3)
+#define ENAOB01_FLAG_CONVERSATION_OVER (1 << 4)
+#define ENAOB01_FLAG_STARTED_RACE (1 << 5)
+#define ENAOB01_FLAG_LAUGH (1 << 6)
+#define ENAOB01_FLAG_SURPRISE (1 << 7)
+#define ENAOB01_FLAG_PLAYER_CAN_TALK (1 << 8)
 
 typedef struct EnAob01 {
     /* 0x000 */ Actor actor;
     /* 0x144 */ EnAob01ActionFunc actionFunc;
     /* 0x148 */ SkelAnime skelAnime;
     /* 0x18C */ ColliderCylinder collider;
-    /* 0x1D8 */ Path* unk_1D8[14];
-    /* 0x210 */ u16 unk_210;
+    /* 0x1D8 */ Path* dogPaths[14];
+    /* 0x210 */ u16 textId;
     /* 0x212 */ Vec3s jointTable[MAMAMU_YAN_LIMB_MAX];
     /* 0x272 */ Vec3s morphTable[MAMAMU_YAN_LIMB_MAX];
-    /* 0x2D2 */ u16 unk_2D2;
+    /* 0x2D2 */ u16 stateFlags;
     /* 0x2D4 */ Vec3s trackTarget;
     /* 0x2DA */ Vec3s headRot;
     /* 0x2E0 */ Vec3s torsoRot;
-    /* 0x2E6 */ Vec3s unk_2E6;
-    /* 0x2EC */ Vec3s unk_2EC;
-    /* 0x2F2 */ Vec3s unk_2F2;
-    /* 0x2F8 */ s16 unk_2F8[MAMAMU_YAN_LIMB_MAX];
-    /* 0x318 */ s16 unk_318[MAMAMU_YAN_LIMB_MAX];
+    /* 0x2E6 */ Vec3s prevTrackTarget;
+    /* 0x2EC */ Vec3s prevHeadRot;
+    /* 0x2F2 */ Vec3s prevTorsoRot;
+    /* 0x2F8 */ s16 limbRotTableY[MAMAMU_YAN_LIMB_MAX];
+    /* 0x318 */ s16 limbRotTableZ[MAMAMU_YAN_LIMB_MAX];
     /* 0x338 */ UNK_TYPE1 unk338[0xB6];
     /* 0x3EE */ s16 eyeIndex;
     /* 0x3F0 */ s16 blinkTimer;
-    /* 0x3F4 */ Actor* unk_3F4;
-    /* 0x3F8 */ Actor* unk_3F8[14];
+    /* 0x3F4 */ Actor* dogToFollow;
+    /* 0x3F8 */ Actor* dogs[14];
```

Comment:
In fact, instead of `ARRAY_COUNT(sDogInfo)`, should you just have a `#define RACEDOG_COUNT 14`, 
```suggestion
    /* 0x3F8 */ Actor* dogs[RACEDOG_COUNT];
```
Then you can do: 
`static DogInfo sDogInfo[RACEDOG_COUNT] = {`
`ARRAY_COUNT(sDogInfo)` -> `RACEDOG_COUNT`
`ARRAY_COUNT(this->dogs)` -> `RACEDOG_COUNT`
More related to this on my next comment